proto/pocket/application/application.proto

```diff
@@ -5,12 +5,14 @@ option go_package = "pocket/x/application/types";
 
 import "cosmos_proto/cosmos.proto";
 import "cosmos/base/v1beta1/coin.proto";
+import "gogoproto/gogo.proto";
+import "google/protobuf/any.proto";
 
 // Application defines the type used to store an on-chain definition and state for an application
 message Application {
   string address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"]; // The Bech32 address of the application using cosmos' ScalarDescriptor to ensure deterministic deterministic encoding using cosmos' ScalarDescriptor to ensure deterministic deterministic encoding
   cosmos.base.v1beta1.Coin stake = 2; // The total amount of uPOKT the application has staked
   // TODO(@Olshansk): Uncomment the line below once the `ServiceId` proto is defined
   // repeated service.ServiceId service_ids = 3; // The ID of the service this session is servicing
+  repeated google.protobuf.Any delegateePubKeys = 4 [(cosmos_proto.accepts_interface) = "cosmos.crypto.PubKey", (gogoproto.nullable) = false]; // The Gateway's corresponding cosmos.crypto.PubKey (interface) encoded into an cosmos.codec.Any for use in the non nullable slice of delegatees the application is delegated to.
```

Comment:
Want to raise a few points:

1. Allow this to be nullable (i.e. you don't have to delegate)
2. Use a repeated string of addresses instead of repeated pubkeys (shorter & simpler) + you have access to the account keeper
3. s/delegateePubKeys/gatewayPubKeys or s/delegateePubKeys/gatewayAddresses so its easier to read and update the comment to explained its delegated to gateways

Reply:
> Allow this to be nullable (i.e. you don't have to delegate)

Nullable causes serialisation errors. The null form of this is an empty array. We want this always present so that the app can sign its own relays off chain and the servicers can build the ring easily (ring only with app pub key). Introducing nullable slice here caused issues in the alpha repo.

> Use a repeated string of addresses instead of repeated pubkeys (shorter & simpler) + you have access to the account keeper

I can do this, as long as the off chain actors can rebuild the ring from the addresses (also has access to the account keeper) this shouldnt be a problem

> s/delegateePubKeys/gatewayPubKeys or s/delegateePubKeys/gatewayAddresses so its easier to read and update the comment to explained its delegated to gateways

makes sense 👍🏼

Reply:
@Olshansk on second thought, as we spoke yesterday offline about not auto-undelegating when a Gateway unstakes, this would make the retrieval of the gateway's public key from their address potentially impossible (if it doesn't exist on chain we cant find the pubkey) and thus the ring cannot be built.

The `Any` type used by cosmos is a customised version of the Any protobuf message that is well supported and this is the standard way of encoding public keys in a protobuf. I would push to keep public keys until this can be worked around

Reply:
Also I went with `DelegateeGatewayPubKey` to be verbose as hell

Reply:
Resolving this (and will start a new thread if needed) per the numerous offline discussions.